services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -186,14 +194,34 @@ async def remove_services_and_data():
     fire_and_forget_task(remove_services_and_data())
 
 
-## PROJECT NODES -----------------------------------------------------
+@observe(event="SIGNAL_USER_DISCONNECTED")
+async def user_disconnected(
+    user_id: int, client_session_id: str, app: web.Application
+) -> None:
+    # check if there is a project resource
+    with managed_resource(user_id, client_session_id, app) as rt:
```

Comment:
``user_disconnected`` is a slot to a signal ``SIGNAL_USER_DISCONNECTED`` emitted from a ``managed_resource`` context manager as well. Question: does this context manager supports nested calls or should you pass the caller's context to this function?

Reply:
it's not nested.